sdk/servicebus/service-bus/review/service-bus.api.md

```diff
@@ -502,6 +503,13 @@ export interface ServiceBusReceiverOptions {
     subQueueType?: "deadLetter" | "transferDeadLetter";
 }
 
+// @public
+export interface ServiceBusRuleManager {
+    createRule(ruleName: string, filter: SqlRuleFilter | CorrelationRuleFilter, ruleAction?: SqlRuleAction, options?: OperationOptionsBase): Promise<void>;
+    deleteRule(ruleName: string, options?: OperationOptionsBase): Promise<void>;
+    getRules(options?: OperationOptionsBase): Promise<RuleProperties[]>;
```

Comment:
do we have to worry about paginating this or will the max collection size always be small enough to fit in a single request?

Reply:
Yeah we discussed offline that we should be use paging (will need to update .NET implementation/API). It should be similar to the admin client method that lists rules.

Reply:
@JoshLove-msft it doesn't look like the response contain next link/continuation information, so we will have to conjure some magic in the convenience layer?

Reply:
https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-request-response#response-11

Reply:
Correct, the response doesn't contain a continuation token, but this is also true for the ATOM API. The way that we can do it is to just keep track of the number of elements returned in each call, and use that to inform the "skip" value that we pass for the next call.

Reply:
In JS we can return an array of T and attach the continuation token to the array

Reply:
> the response doesn't contain a continuation token, but this is also true for the ATOM API.
our JS ATOM code seems to be parsing a nextLink to retrieve the `$skip` query parameter from the response
https://github.com/jeremymeng/azure-sdk-for-js/blob/81a37f146cdee62c76ab299d97d411f3aa5d1f6b/sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts#L2846

Reply:
Interesting.. we don't actually use that in . NET - we calculate it on the client.

Reply:
> In JS we can return an array of T and attach the continuation token to the array

We changed this recently in TA because the docs for a custom array type look odd: https://docs.microsoft.com/en-us/javascript/api/@azure/ai-text-analytics/detectlanguageresultarray?view=azure-node-latest

I think there was also a performance degradation as soon as you place an extra property on an array type. /cc @deyaaeldeen @witemple-msft to check my memory on this.

Reply:
> We changed this recently in TA because the docs for a custom array type look odd: https://docs.microsoft.com/en-us/javascript/api/@azure/ai-text-analytics/detectlanguageresultarray?view=azure-node-latest

That is correct. TA v5 attached various things to array objects such as statistics and model version information. In the newer version, I am no longer exposing any of that and advising customers to use the `onResponse` callback to access it on the raw response. For continuation tokens, TA never exposed them, and the newer version continues to not expose them because I never heard about customers asking for it.

Reply:
On a side note, please consider using the core-paging primitive if you're implementing a pageable operation: https://github.com/Azure/azure-sdk-for-js/blob/1ba1578477385aa921e8cbb44061e1a0311414cd/sdk/core/core-paging/src/getPagedAsyncIterator.ts#L13-L20

Reply:
I removed continuationToken support and just return array of rules for page. We have used this pattern a lot in JS SDK.

> On a side note, please consider using the core-paging primitive if you're implementing a pageable operation

Great point! I will do that after this beta release.

Reply:
logged https://github.com/Azure/azure-sdk-for-js/issues/22108 to track moving to use core-paging primitive.